### PR TITLE
Update theme.json docs with new properties added

### PR DESCRIPTION
**Pull request by oandregal**
This PR adds lacking documentation for some new properties that were added to the style system ([font family](https://github.com/WordPress/gutenberg/pull/24868), [font style](https://github.com/WordPress/gutenberg/pull/26444), [font weight](https://github.com/WordPress/gutenberg/pull/26444), [text decoration](https://github.com/WordPress/gutenberg/pull/26059), [text transform](https://github.com/WordPress/gutenberg/pull/26060)) which landed without proper docs.

### CODE REVIEW

File: docs/designers-developers/developers/themes/theme-json.md [hunk 1]
```diff
@@ -299,42 +308,53 @@ These are the current color properties supported by blocks:
...
-| Post Comments | Yes | Yes |
-| Post Comments Count | Yes | Yes |
-| Post Comments Form | Yes | Yes |
-| Post Date | Yes | Yes |
-| Post Excerpt | Yes | Yes |
-| Post Tags | Yes | Yes |
-| Post Title | Yes | Yes |
-| Site Tagline | Yes | Yes |
-| Site Title | Yes | Yes |
+| Context | Font Family | Font Size | Font Style | Font Weight | Line Height | Text Decoration | Text Transform |
```
Comment (kjellr) [inline docs/designers-developers/developers/themes/theme-json.md:341]: I think the markup for this isn't correct — it's not rendering in the preview: 

Screen Shot 2020-11-11 at 9 14 25 AM

Comment (nosolosw) [reply docs/designers-developers/developers/themes/theme-json.md:341]: Good catch, thanks! It's fixed as of https://github.com/WordPress/gutenberg/pull/26891/commits/843e2fad3490268dded4cac54746ed3606e8a680

**Review by jorgefilipecosta — APPROVED** [review]
LGTM 👍

**Comment by aristath** [comment]
I'm reading the docs, trying to follow along and get these things to work in an FSE theme and I don't see the controls added anywhere. Is there an example theme somewhere with these things working? Should they be in global and/or the navigation block? :thinking: 
Docs look technically good

**Pull request merged**